flink-connectors/flink-hbase/src/main/java/org/apache/flink/addons/hbase/HBaseLookupFunction.java

```diff
@@ -0,0 +1,150 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.addons.hbase;
+
+import org.apache.flink.addons.hbase.util.HBaseConfigurationUtil;
+import org.apache.flink.addons.hbase.util.HBaseReadHelper;
+import org.apache.flink.annotation.VisibleForTesting;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.table.functions.FunctionContext;
+import org.apache.flink.table.functions.TableFunction;
+import org.apache.flink.types.Row;
+import org.apache.flink.util.StringUtils;
+
+import org.apache.hadoop.conf.Configuration;
+import org.apache.hadoop.hbase.HBaseConfiguration;
+import org.apache.hadoop.hbase.HConstants;
+import org.apache.hadoop.hbase.TableName;
+import org.apache.hadoop.hbase.TableNotFoundException;
+import org.apache.hadoop.hbase.client.Connection;
+import org.apache.hadoop.hbase.client.ConnectionFactory;
+import org.apache.hadoop.hbase.client.HTable;
+import org.apache.hadoop.hbase.client.Result;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+
+/**
+ * The HBaseLookupFunction is a standard user-defined table function, it can be used in tableAPI
+ * and also useful for temporal table join plan in SQL.
+ */
+public class HBaseLookupFunction extends TableFunction<Row> {
+	private static final Logger LOG = LoggerFactory.getLogger(HBaseLookupFunction.class);
+	private static final long serialVersionUID = 1L;
+
+	private final String hTableName;
+	private final byte[] serializedConfig;
+	private final HBaseTableSchema hbaseTableSchema;
+
+	private transient HBaseReadHelper readHelper;
+	private transient Connection hConnection;
+	private transient HTable table;
+
+	public HBaseLookupFunction(
+			Configuration configuration,
+			String hTableName,
+			HBaseTableSchema hbaseTableSchema) {
+		this.serializedConfig = HBaseConfigurationUtil.serializeConfiguration(configuration);
+		this.hTableName = hTableName;
+		this.hbaseTableSchema = hbaseTableSchema;
+	}
+
+	/**
+	 * The invoke entry point of lookup function.
+	 * @param rowKey the lookup key. Currently only support single rowkey.
+	 */
+	public void eval(Object rowKey) throws IOException {
+		// fetch result
+		Result result = table.get(readHelper.createGet(rowKey));
+		if (!result.isEmpty()) {
+			// parse and collect
+			collect(readHelper.parseToRow(result, rowKey));
+		}
+	}
+
+	@Override
+	public TypeInformation<Row> getResultType() {
+		return hbaseTableSchema.convertsToTableSchema().toRowType();
+	}
+
+	private org.apache.hadoop.conf.Configuration prepareRuntimeConfiguration() throws IOException {
```

Comment:
No actual exception thrown, consider remove the `throws` declaration from method signature.